Hassan_Ali/Week3_binary_tree/are_they_cousins.py

```diff
@@ -0,0 +1,91 @@
+class TreeNode:
+    def __init__(self,val:str):
+        self.val = val
+        self.left = None
+        self.right = None
+
+#using recursion
+
+def are_they_cousins(root:TreeNode,person1:str,person2:str)->bool:
+
+    p1_parent = None
+    p1_depth = 0
+    p1_found = False
+
+    p2_parent = None
+    p2_depth = 0
+    p2_found = False
+
+    # using a helper function to keep track of the parent and depth of the tree
+
+    def find_cousins(node:TreeNode,parent:TreeNode,depth:int)->bool:
+        if not node:
+            return False
+
+        if person1 == node.val:
+           nonlocal p1_found 
+           p1_found = True
+           nonlocal p1_depth 
+           p1_depth = depth
+           nonlocal p1_parent
+           p1_parent = parent
+
+        if person2 == node.val:
+            nonlocal p2_found
+            p2_found = True
+            nonlocal p2_depth
+            p2_depth = depth
+            nonlocal p2_parent
+            p2_parent = parent
```

Comment:
A lot of repetitive code, but your solutions works.

Reply:
> A lot of repetitive code, but your solutions works.

I have tried to remove the repetitive declaration and initialization of the variables.